test/config/test_configurable.rb

```diff
@@ -346,6 +346,19 @@ class BufferChild
       configured_in :buffer
       config_param :size_of_something, :size, default: 128
     end
+
+    class BufferBase
+      include Fluent::Configurable
+    end
+
+    class BufferSubclass < BufferBase
+      attr_accessor :owner
+      configured_in :buffer
+      config_param :size_of_something, :size, default: 512
+    end
+
+    class BufferSubclass2 < BufferSubclass
```

Comment:
`BufferSubclass2` looks like same hierarchy with `BufferSubclass`.
`BufferSubSubclass` or other name is better.